src/main/scala/tech/cryptonomic/conseil/config/LorreAppConfig.scala

```diff
@@ -61,60 +91,139 @@ trait LorreAppConfig {
-  protected def loadApplicationConfiguration(
-      commandLineArgs: Array[String]
-  ): ConfigReader.Result[CombinedConfiguration] = {
-
-    /** Use the pureconfig convention to handle configuration from the command line */
-    def readArgs(args: Array[String]): ConfigReader.Result[ArgumentsConfig] =
-      argsParser.parse(args, ArgumentsConfig()).toRight[ConfigReaderFailures](sys.exit(1))
-
-    //applies convention to uses CamelCase when reading config fields
-    @silent("local method hint in method loadApplicationConfiguration is never used")
-    implicit def hint[T]: ProductHint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
-    @silent("local val depthHint in method loadApplicationConfiguration is never used")
-    implicit val depthHint: EnumCoproductHint[Depth] = new EnumCoproductHint[Depth]
-    @silent("local val chainEventHint in method loadApplicationConfiguration is never used")
-    implicit val chainEventHint = new FieldCoproductHint[ChainEvent]("type") {
-      override def fieldValue(name: String): String = name.head.toLower +: name.tail
-    }
-
-    val loadedConf = for {
-      args <- readArgs(commandLineArgs)
-      ArgumentsConfig(depth, verbose, headHash, network) = args
-      lorre <- loadConfig[LorreConfiguration](namespace = "lorre").map(_.copy(depth = depth, headHash = headHash))
-      nodeRequests <- loadConfig[NetworkCallsConfiguration]("lorre")
-      node <- loadConfig[TezosNodeConfiguration](namespace = s"platforms.tezos.$network.node")
-      streamingClient <- loadAkkaStreamingClientConfig(namespace = "akka.tezos-streaming-client")
-      fetching <- loadConfig[BatchFetchConfiguration](namespace = "lorre.batchedFetches")
-    } yield
-      CombinedConfiguration(
-        lorre,
-        TezosConfiguration(network, node),
-        nodeRequests,
-        streamingClient,
-        fetching,
-        VerboseOutput(verbose)
-      )
+  private val loadApplicationConfiguration: Kleisli[IO, List[String], CombinedConfiguration] = Kleisli {
+    commandLineArgs =>
+      import cats.instances.either._
+
+      lazy val failuresPolicy = {
+        val envConf = sys.env.get(LORRE_FAILURE_IGNORE_VAR)
+        ProcessingFailuresPolicy(
+          varName = LORRE_FAILURE_IGNORE_VAR,
+          ignore = envConf.exists(ignore => ignore.toLowerCase == "true" || ignore.toLowerCase == "yes")
+        )
+      }
+
+      /** Use the pureconfig convention to handle configuration from the command line */
+      def readArgs(args: List[String]): ConfigReader.Result[ArgumentsConfig] =
+        argsParser
+          .parse(args, ArgumentsConfig())
+          .toRight[ConfigReaderFailures](ConfigReaderFailures(new ThrowableFailure(new IllegalArgumentException, None)))
+
+      //applies convention to uses CamelCase when reading config fields
+      @silent("local method hint in method loadApplicationConfiguration is never used")
```

Comment:
because you changed `loadApplicationConfiguration` to val  `@silent` annotations are not working, please fix them

Reply:
Does the annotation only work on `def`s?

Reply:
no, the warning is phrased differently - there is `value` instead of `method` and `$anonfunction`(if I remember correctly) instead of `loadApplicationConfiguration`